WooCommerce/Classes/Extensions/String+Helpers.swift

```diff
@@ -68,4 +68,150 @@ extension String {
+        ///
+        func replaceUnderscoreDashAndPlusWithDot(inString: String) -> String {
+            var outString = inString
+            let replaceTargets = ["_", "-", "+"]
+            for replaceTarget in replaceTargets {
+                outString = outString.replacingOccurrences(of: replaceTarget, with: ".")
+            }
+            return outString
+        }
+
+        /// Insert a . before and after any non number
+        ///
+        func insertDotsBeforeAndAfterAnyNonNumber(inString: String) -> String {
+            guard inString.count > 1 else {
+                return inString
+            }
+
+            /// Initialize our output with the first character of the input string
+            var outString = String(inString[inString.index(inString.startIndex, offsetBy: 0)])
+
+            /// Loop over the remaining characters in the string, inserting . as needed
+            for index in 1...inString.count - 1 {
+                let characterAtIndex = inString[inString.index(inString.startIndex, offsetBy: index)]
+                let characterBefore = inString[inString.index(inString.startIndex, offsetBy: index - 1)]
+
+                let characterAtIndexIsNumberOrDot = characterAtIndex.isNumber || characterAtIndex == "."
+                let characterBeforeIsNumberOrDot = characterBefore.isNumber || characterBefore == "."
+
+                if !characterAtIndexIsNumberOrDot && characterBefore.isNumber {
+                    outString += "."
+                }
+
+                if characterAtIndex.isNumber && !characterBeforeIsNumberOrDot {
+                    outString += "."
+                }
+
+                outString += String(characterAtIndex)
+            }
+
+            return outString
+        }
+
+        /// Ranked per https://www.php.net/manual/en/function.version-compare.php
+        ///
+        enum ComponentScore: Comparable {
+            case other
+            case dev
+            case alpha
+            case beta
+            case RC
+            case number
+            case patch
+        }
+
+        /// Score each component
+        ///
+        func scoreComponent(stringComponent: String) -> ComponentScore {
+            if stringComponent == "dev" {
+                return .dev
+            }
+            if stringComponent == "alpha" || stringComponent == "a" {
+                return .alpha
+            }
+            if stringComponent == "beta" || stringComponent == "b" {
+                return .beta
+            }
+            if stringComponent == "RC" || stringComponent == "rc" {
+                return .RC
+            }
+            let componentCharacterSet = CharacterSet(charactersIn: stringComponent)
+            if componentCharacterSet.isSubset(of: .decimalDigits) {
+                return .number
+            }
+
+            if stringComponent == "pl" || stringComponent == "p" {
+                return .patch
+            }
+
+            return .other
+        }
```

Comment:
nit: these could be in the enum's initializer:

```suggestion
    extension ComponentScore {
        func init(stringComponent: String) {
            if stringComponent == "dev" {
                return self = .dev
            }
            if stringComponent == "alpha" || stringComponent == "a" {
                return self = .alpha
            }
            if stringComponent == "beta" || stringComponent == "b" {
                return self = .beta
            }
            if stringComponent == "RC" || stringComponent == "rc" {
                return self = .RC
            }
            let componentCharacterSet = CharacterSet(charactersIn: stringComponent)
            if componentCharacterSet.isSubset(of: .decimalDigits) {
                return self = .number
            }

            if stringComponent == "pl" || stringComponent == "p" {
                return self = .patch
            }

            self = .other
        }
```

Reply:
Love it! Fixed in https://github.com/woocommerce/woocommerce-ios/pull/5382/commits/0a631e38f0c3b5824438e11df9e2743d3af30363